### PR TITLE
Decorator class

### PR DESCRIPTION
**Pull request by Tamunokuro**
- Create nameable class
- Create BaseDecorator class
- Create CapitalizeDecorator class
- Create TrimmerDecorator class

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
+require_relative 'nameable'
+
+class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
 
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
     @id = Random.rand(1..1000)
     @name = name
```
Comment (ghost) [inline person.rb:40]: - [ ] When I try to test your application, I get this error. Kindly fix it

image

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @Tamunokuro , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Correct use of Gitflow
✅ No linter error


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,41 @@
...
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    name = @nameable.correct_name
+    name.length > 10 ? name[0..9] : name
+  end
```
Comment (omar-muhamad) [inline person.rb:30]: - [ ] [Optional] [Highly Recommended] It would be better professionally if add each of these classes in a separate module.

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @Tamunokuro,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### ⚠️Please consider the Optional suggestions below⚠️



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- Please check the inline comments below ⬇.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**